### PR TITLE
implements --require-pragma and --insert-pragma

### PR DESCRIPTION
**Pull request by roippi**
Closes https://github.com/prettier/plugin-php/issues/454.

### CODE REVIEW

File: tests/insert-pragma/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+echo "Hello World";
+
+// non-docblock comment
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+<div>
+  inline stuff up top
+</div>
+
+<?php /** @format */
```
Comment (evilebottnawi) [inline tests/insert-pragma/__snapshots__/jsfmt.spec.js.snap]: Not sure it is good idea do this, maybe better insert new php tag at the top of file with pragma, but we can do this after solve problem with inline printing

Comment (roippi) [reply tests/insert-pragma/__snapshots__/jsfmt.spec.js.snap]: Agree, I wasn't thrilled with doing it this way

Comment (roippi) [reply tests/insert-pragma/__snapshots__/jsfmt.spec.js.snap]: I think I will just drop `--insert-pragma` support for files that don't start with a php tag.  Seems simplest.

File: src/pragma.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
+"use strict";
+
+const parse = require("./parser");
+const { EOL } = require("os");
+
+const extractDocBlocks = text => {
+  const parsed = parse(text);
```
Comment (mgrip) [inline src/pragma.js]: any idea how often this gets called or if there's any performance concern? (I'm guessing once per file which probably isn't a huge deal)

Comment (roippi) [reply src/pragma.js]: Yup, just checked:

- once per file, if you use no flags or `--require-pragma`
- twice per file, if you use `--insert-pragma` (or both flags)

Some room for optimization.  I didn't know that this hook is still called even when you use no flags.

Comment (roippi) [reply src/pragma.js]: I'll do a couple quick optimizations here, give me a bit.

Comment (roippi) [reply src/pragma.js]: Test suite slowed down by ~1% with my original patch, now closer to .1%.  Pretty much noise.

LMK if you don't want the new npm dependency, memoize impls are pretty interchangeable.

**Comment by evilebottnawi** [comment]
@mgrip need minor release

**Pull request merged**